### PR TITLE
Dynamic Group of Groups: Add draggable child groups, recompute and hide weight.

### PR DESCRIPTION
**Pull request by bryanculver**
# Closes: #DNE
# What's Changed

- Add dragging and re-weighting to Dynamic Group of Groups memberships

**Drag to Re-Weight**

https://user-images.githubusercontent.com/31187/179160540-bc9face5-8d3d-4d13-859d-a5dfba84c8a9.mov

**Ignores Empty Rows**

https://user-images.githubusercontent.com/31187/179161313-12f5bcee-772f-4b5e-a018-7f01c2c89ed2.mov

**

# TODO

- [x] Explanation of Change(s)
- [x] Attached Screenshots, Payload Example
- NA Unit, Integration Tests
- NA Documentation Updates (when adding/changing features)
- NA Example Plugin Updates (when adding/changing features)
- NA Outline Remaining Work, Constraints from Design

### CODE REVIEW

**Comment by bryanculver** [comment]
Screenshot above still had some cache from #1710. Re-recorded:

https://user-images.githubusercontent.com/31187/179193173-a6f9f292-b291-4afe-9130-ffada2d3dc7a.mov

File: nautobot/extras/templates/extras/dynamicgroup_edit.html [hunk 1]
```diff
@@ -62,13 +62,15 @@
                                 {% if forloop.first %}
                                     <thead>
                                         <tr>
+                                            <th>Sort</th>
```
Comment (jathanism) [inline nautobot/extras/templates/extras/dynamicgroup_edit.html:65]: I'm thinking we might still want to call this column **Weight** or something that is indicative that ordering is significant?

File: nautobot/extras/templates/extras/dynamicgroup_edit.html [hunk 2]
```diff
@@ -112,5 +114,36 @@
...
         added: jsify_form
     });
+
+    // Function that takes a Dynamic Group children form and recomputes the weights.
+    function update_weights(children_form){
+        let w = 10;
+        $(children_form).find('.formset_row-{{ children.prefix }}').each((i, el) => {
+            let contains_value = $(el).find("select[name$='-group'], select[name$='-operator']").map((idx, sel) => {
+                return ($(sel).val().length > 0)
+            }).toArray().some(x => x === true)
```
Comment (jathanism) [inline nautobot/extras/templates/extras/dynamicgroup_edit.html:124]: I can grok that this appears to be arriving at the terminal value of `.some()` (jQuery's analog to Python's `any()`), but would you mind adding a brief comment to explain the intent of this block?

File: nautobot/extras/templates/extras/dynamicgroup_edit.html [hunk 3]
```diff
@@ -112,5 +114,36 @@
...
+
+            let weight_input = $(el).find("*[name$='-weight']")
+            weight_input.val("")
+            if(contains_value) {
+                weight_input.val(w);
+                w+=10;
```
Comment (jathanism) [inline nautobot/extras/templates/extras/dynamicgroup_edit.html:130]:
```suggestion
                w += 10;
```

**Review by jathanism — CHANGES_REQUESTED** [review]
Just some minor nits but I'm loving this. Thank you!

**Pull request merged**